src/js/view/view.js

```diff
@@ -875,15 +875,29 @@ function View(_api, _model) {
 
             // Resize within MAX_FLOATING_WIDTH×MAX_FLOATING_HEIGHT bounds, never enlarge.
             const { width, height } = _this.getSafeRegion(false);
-            const ratio = Math.min(1, MAX_FLOATING_WIDTH / width, MAX_FLOATING_HEIGHT / height);
-            _this.resize(width * ratio, height * ratio, true);
+            const floatingWidth = Math.min(width, MAX_FLOATING_WIDTH);
+            const floatingHeight = Math.min(height * floatingWidth / width, MAX_FLOATING_HEIGHT);
+            _this.resize(floatingWidth, floatingHeight);
 
             _resizeOnFloat = false;
         } else if (isVisible) {
             _this.stopFloating();
         }
     }
 
+    this.resizeFloatingPlayer = function() {
+        if (floatingPlayer && !_resizeOnFloat) {
+            const rect = bounds(_playerElement);
+            const containerWidth = Math.round(rect.width);
+            const containerHeight = Math.round(rect.height);
+            const floatingWidth = Math.min(containerWidth, MAX_FLOATING_WIDTH);
```

Comment:
We could make this DRYer by reusing part of `_updateFloating`:
```
const { width, height } = _this.getSafeRegion(false);
const floatingWidth = Math.min(width, MAX_FLOATING_WIDTH);
const floatingHeight = Math.min(height * floatingWidth / width, MAX_FLOATING_HEIGHT);
```
we definitely want our floatingHeight and floatingWidth converters to be in one place